cs/src/core/Allocator/GenericAllocator.cs

```diff
@@ -191,6 +191,13 @@ internal override bool TryComplete()
         /// </summary>
         public override void Dispose()
         {
+            // Dispose base allocator
+            base.Dispose();
+
+            // Ensure pending IO on object log device is completed
+            objectLogDevice.CompletePending();
```

Comment:
This is probably a left over issue from before, but why does Dispose() here not call into dispose of objectLogDevice? It seems wrong to call CompletePending() without disposing, as that does not reliable stop future calls.

Reply:
We cannot dispose the device as we did not create the device, the user did. So they will have to dispose it.